.github/workflows/performance.yml

```diff
@@ -0,0 +1,151 @@
+# This is a workflow triggered by PR or triggered manually
+# Runs quick performance tests and reports the comparison against HEAD
+# Test should take less than 10 minutes to run on current self-hosted devices
+name: "Performance Testing"
+
+# Controls when the action will run.
+# Workflow runs when manually triggered using the UI or API.
+on:
+  push:
+  pull_request:
+    branches: [ main ]
+
+# Env variables
+env:
+  SG_COMMIT: 649509c
+
+jobs:
+  Performance_x86-64:
+    name: Performance x86-64
+    runs-on: [self-hosted, linux, x64]
+    # Inputs the workflow accepts.
+    steps:
+      - run: echo "This job is now running on a ${{ runner.os }} self-hosted server."
+
+      - name: Setup Rust Toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+
+      - name: Build sightglass commit ${{env.SG_COMMIT}}"
+        run: |
+          cd ../ && ls -l && rm -rf ./sightglass
+          git clone https://github.com/bytecodealliance/sightglass.git && cd ./sightglass
+          git checkout ${{env.SG_COMMIT}}
+          cargo build --release
+
+      - name: Checkout ${{ github.ref }} from ${{ github.repository }}
+        uses: actions/checkout@v2
+        with:
+          submodules: true
+          path: wasmtime_commit
+
+      - name: Build ${{ github.ref }}
+        working-directory: ./wasmtime_commit
+        run: |
+          cargo build --release -p wasmtime-bench-api
+          cp target/release/libwasmtime_bench_api.so /tmp/wasmtime_commit.so
+
+      - name: Checkout Main
+        uses: actions/checkout@v2
+        with:
+          ref: 'main'
+          submodules: true
+          path: wasmtime_main
```

Comment:
When a push to `main` happens I think this'll double-checkout main and build it twice?

Ideally though it'd be awesome if we could do just one build as part of this PR and use stored data for previous runs perhaps for comparison. We should be guaranteed that all commits on `main` have performance information so comparing a new merge to main or a PR is ideally just about picking the revision to compare against.

Reply:
To be honest, I think there is some caching that is taking place via this github code that lives on the server. I do though the blast the sightglass directory and rebuild that. I don't blast the wasmtime directories but at some point during developing this it seemed those directories were removed but then very quickly rebuilt by the environment, implying to me that something was cached. 

This iteration doesn't actually test against much of anything (that will be critiqued for improvement), but as a result this entire process from start to finish only takes about 4-1/2 minutes, so even if everything is rebuilt it currently really doesn't take that long. Also like I mentioned above (and suggested offline), I want this to only run on demand not after every PR is submitted or updated. For this patch, I wanted to have something that was for sure going to work every time and then get fancy with the caching and other efficiency improvements on a future iteration. If there is something obvious to change now though, lets do it.

Reply:
Ah ok if it's fast enough then seems reasonable to at least start with this and we can alter later if it ever becomes necessary